git/commits_test.go

```diff
@@ -137,8 +150,13 @@ func TestGetConventionalCommitTypesSinceLatestRelease(t *testing.T) {
 		}
 
 		assert.NoError(t, err)
-		assert.Equal(t, test.expectedLastVersion, actual.LatestReleaseVersion)
+
+		// We need to compare using semver's own Equal function, because under
+		// the hood it makes a difference between versions with a leading v and
+		// versions without a leading v. However, when printing them, they are
+		// actually both shown without the v.
+		assert.True(t, test.expectedLastVersion.Equal(actual.LatestReleaseVersion))
+		assert.Equal(t, test.expectedLastVersion.String(), actual.LatestReleaseVersion.String())
```

Comment:
Do we need both of these asserts?  They should be equal, right? I prefer the first one of the two.

Reply:
I thought that it may actually help to have both, because otherwise we only rely on the `Equal` methof of `SemVer` with its special handling of `v` values (that's why I also left the comment, to make things more clear here).

Reply:
It does not handle `v` special though. The equal check totally ignores it, which is what we want here I think.

Reply:
That's what I called special, since it does *not* check for equality! `v1.0.0` and `1.0.0` are not equal, but treated as such.

Reply:
Yeah that is a possible point of view. I would have said that from a versioning point of view they are the same version. Therefore ignoring the `v` is what I expected. But it might not be as intuitive for everybody.

Reply:
I see. In the end, we have two types of equality here:

- From a semantic point of view, they are equal.
- From a technical point of view, they aren't.

That's why I had implemented both assertions, first to ensure that they are semantically equal, but then also a technical check to avoid relying on a function whose behavior may change in the future (without noticing it).

So, what should we do now in the end?

Reply:
I understand the intend and agree with it now. I still think that using `.String()` might be the wrong way to implement this technical check. `.String()` is also part of that library and could change (to example always omit the `v`) as well. I don't see why this is less likely than the behavior of   `.Equal()` changing. Maybe there is an other way of implementing this technical check? Maybe use `assert.Equal` (like the deleted check) and `version.Equal()` (like the first of the new ones) instead?

Reply:
Hm, yes, that's true …

Unfortunately, the original check failed, because they are different – it correctly detected the difference between the version with and without the leading `v` 🙈

Reply:
I've now followed your suggestion, removed one ofthe two assertions, and adjusted the comment.